tests/test_state/test_prosumption_interface.py

```diff
@@ -0,0 +1,125 @@
+        assert isinstance(prosumption_interface.get_forecast_measurement_deviation_kWh(
+            time_slot=current_time_slot), float)
+        assert isinstance(prosumption_interface.get_unsettled_deviation_kWh(
+            time_slot=current_time_slot), float)
+
+    @staticmethod
+    def _setup_configuration_for_settlement_posting(
+            energy_deviation=None, unsettled_deviation=None):
+        prosumption_interface = ProsumptionInterfaceHelper()
+        current_time_slot = now()
+        if energy_deviation:
+            prosumption_interface._forecast_measurement_deviation_kWh[
+                current_time_slot] = energy_deviation
+        if unsettled_deviation:
+            prosumption_interface._unsettled_deviation_kWh[
+                current_time_slot] = unsettled_deviation
+        return prosumption_interface, current_time_slot
+
+    @pytest.mark.parametrize(
+        "energy_deviation, expected_response", [(None, False), (-1, False), (1, True)])
+    def test_can_post_settlement_bid_return_expected(self, energy_deviation, expected_response):
+        prosumption_interface, time_slot = self._setup_configuration_for_settlement_posting(
+            energy_deviation)
+        assert prosumption_interface._forecast_measurement_deviation_kWh.get(
+            time_slot) is energy_deviation
+        assert prosumption_interface.can_post_settlement_bid(
+            time_slot) == expected_response
+
+    @pytest.mark.parametrize(
+        "energy_deviation, expected_response", [(None, False), (-1, True), (1, False)])
+    def test_can_post_settlement_offer_return_expected(self, energy_deviation, expected_response):
+        prosumption_interface, time_slot = self._setup_configuration_for_settlement_posting(
+            energy_deviation)
+        assert prosumption_interface._forecast_measurement_deviation_kWh.get(
+            time_slot) is energy_deviation
+        assert prosumption_interface.can_post_settlement_offer(
+            time_slot) == expected_response
+
+    @pytest.mark.parametrize(
+        "energy_deviation, unsettled_deviation, expected_response",
+        [(1, None, None), (None, 1, None), (1, 1, True)])
+    def test_get_signed_unsettled_deviation_kWh_return_expected(
+            self, energy_deviation, unsettled_deviation, expected_response):
+        copysign_mock = MagicMock()
+        with patch("gsy_e.models.state.copysign", return_value=copysign_mock):
+            prosumption_interface, time_slot = self._setup_configuration_for_settlement_posting(
+                energy_deviation=energy_deviation, unsettled_deviation=unsettled_deviation)
+            if expected_response:
+                assert prosumption_interface.get_signed_unsettled_deviation_kWh(
+                    time_slot) == copysign_mock
+            else:
+                assert prosumption_interface.get_signed_unsettled_deviation_kWh(
+                    time_slot) is expected_response
```

Comment:
I think something is wrong here. The outcome of  `get_signed_unsettled_deviation_kWh` should either be a number or None. Here you are checking for None and `copysign_mock`. I would expect this implementation:
```suggestion
        assert prosumption_interface.get_signed_unsettled_deviation_kWh(
            time_slot) == expected_response
```

Reply:
I understand what you mean. 
I agree with the fact that the method returns either None or a number. For the case on when it returns a number, this number is at the same time the return value for `copysign` function (being mocked). 
The condition block is to account for the case in which None value is expected, and when the number (mock) is.
For the sake of clearness I'll split this in two different test cases.